### PR TITLE
feat(argocd-notifications): add service account annotations

### PR DESCRIPTION
**Pull request by cesarokuti**
Signed-off-by: Cesar Okuti <cesar.okuti@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [ ] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Review by romulofranca — APPROVED** [review]

**Review by oliverbaehler — CHANGES_REQUESTED** [review]
Please Bump to Chart Version `1.1.2`

File: charts/argocd-notifications/templates/bots/slack/serviceaccount.yaml [hunk 1]
```diff
@@ -5,4 +5,8 @@ metadata:
   name: {{ include "argocd-notifications.bots.slack.serviceAccountName" . }}
   labels:
     {{- include "argocd-notifications.bots.slack.labels" . | nindent 4 }}
+  {{- if .Values.bots.slack.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml .Values.bots.slack.serviceAccount.annotations | nindent 4 }}
```
Comment (oliverbaehler) [inline charts/argocd-notifications/templates/bots/slack/serviceaccount.yaml:10]:
```suggestion
  {{- with .Values.bots.slack.serviceAccount.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
```

File: charts/argocd-notifications/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -5,4 +5,8 @@ metadata:
   name: {{ include "argocd-notifications.serviceAccountName" . }}
   labels:
     {{- include "argocd-notifications.labels" . | nindent 4 }}
+  {{- if .Values.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml .Values.serviceAccount.annotations | nindent 4 }}
```
Comment (oliverbaehler) [inline charts/argocd-notifications/templates/serviceaccount.yaml:10]:
```suggestion
  {{- with .Values.serviceAccount.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
```

**Review by oliverbaehler — CHANGES_REQUESTED** [review]
Bump Chart Version and Replace `if` statement.

**Review by mkilchhofer — APPROVED** [review]
LGTM

**Pull request merged**